docs/protocols.ipynb

```diff
@@ -181,16 +145,300 @@
+    "|`cirq.decompose_once`| Decomposes a value into operations, if possible. |\n",
+    "|`cirq.decompose_once_with_qubits`| Decomposes a value into operations on the given qubits. |\n",
+    "|`cirq.equal_up_to_global_phase`| Determine whether two objects are equal up to global phase. |\n",
+    "|`cirq.has_channel`| Returns whether the value has a channel representation. |\n",
+    "|`cirq.has_mixture`| Returns whether the value has a mixture representation. |\n",
+    "|`cirq.has_unitary`| Determines whether the value has a unitary effect. |\n",
+    "|`cirq.inverse`| Returns the inverse `val**-1` of the given value, if defined. |\n",
+    "|`cirq.is_measurement`| Determines whether or not the given value is a measurement. |\n",
+    "|`cirq.is_parameterized`| Returns whether the object is parameterized with any Symbols. |\n",
+    "|`cirq.measurement_key`| Get the single measurement key for the given value. |\n",
+    "|`cirq.measurement_keys`| Gets the measurement keys of measurements within the given value. |\n",
+    "|`cirq.mixture`| Return a sequence of tuples representing a probabilistic unitary. |\n",
+    "|`cirq.num_qubits`| Returns the number of qubits, qudits, or qids `val` operates on. |\n",
+    "|`cirq.parameter_names`| Returns parameter names for this object. |\n",
+    "|`cirq.parameter_symbols`| Returns parameter symbols for this object. |\n",
+    "|`cirq.pauli_expansion`| Returns coefficients of the expansion of val in the Pauli basis. |\n",
+    "|`cirq.phase_by`| Returns a phased version of the effect. |\n",
+    "|`cirq.pow`| Returns `val**factor` of the given value, if defined. |\n",
+    "|`cirq.qasm`| Returns QASM code for the given value, if possible. |\n",
+    "|`cirq.qid_shape`| Returns a tuple describing the number of quantum levels of each |\n",
+    "|`cirq.quil`| Returns the QUIL code for the given value. |\n",
+    "|`cirq.read_json`| Read a JSON file that optionally contains cirq objects. |\n",
+    "|`cirq.resolve_parameters`| Resolves symbol parameters in the effect using the param resolver. |\n",
+    "|`cirq.to_json`| Write a JSON file containing a representation of obj. |\n",
+    "|`cirq.trace_distance_bound`| Returns a maximum on the trace distance between this effect's input |\n",
+    "|`cirq.trace_distance_from_angle_list`| Given a list of arguments of the eigenvalues of a unitary matrix, |\n",
+    "|`cirq.unitary`| Returns a unitary matrix describing the given value. |\n",
+    "|`cirq.validate_mixture`| Validates that the mixture's tuple are valid probabilities. |\n",
+    "\n",
+    "\n",
+    "### Quantum operator representation protocols\n",
+    "\n",
+    "The following family of protocols is an important and frequently used set of features of Cirq and it is worthwhile mentioning them and and how they interact with each other. They are, in the order of increasing generality:\n",
+    "\n",
+    "* `*unitary`\n",
```

Comment:
How come the protocols are denoted "*unitary" as opposed to just "unitary" or "cirq.unitary"?

Reply:
Well, with the definition of a protocol above, being a single class defining the magic methods to implement and the helper methods around it, technically has_unitary, apply_unitary and unitary are three separate protocols: SupportsExplicitHasUnitary, SupportsConsistentApplyUnitary, SupportsUnitary. The same applies to the others.